solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -1285,6 +1360,47 @@ contract StandardPoolConverter is
         return _amount.mul(_reserveBalance) / _supply;
     }
 
+    /**
+     * @dev returns two samples required for calculating the network fee
+     *
+     * @param currProd  the current product of the balances
+     * @param prevProd  the previous product of the balances
+     *
+     * @return the square root of the current product of the balances
+     * @return the square root of the previous product of the balances
+     */
+    function networkFeePoints(
+        uint256 currProd,
+        uint256 prevProd
+    ) private pure returns (uint256, uint256) {
+        return (
+            currProd > 0 ? MathEx.floorSqrt(currProd) : 0,
+            prevProd > 0 ? MathEx.floorSqrt(prevProd) : 0
+        );
+    }
+
+    /**
+     * @dev returns the network fee ratio
+     *
+     * @param currPoint     the current product of the balances
+     * @param prevPoint     the previous product of the balances
+     * @param networkFee    the relative portion (in PPM units) that
+     * should be taken from the conversion fees accumulated in this pool
+     *
+     * @return numerator of the portion that should be taken from the balances of this pool
+     * @return denominator of the portion that should be taken from the balances of this pool
+     */
+    function networkFeeRatio(
+        uint256 currPoint,
+        uint256 prevPoint,
+        uint256 networkFee
+    ) private pure returns (uint256, uint256) {
+        return (
+            (currPoint - prevPoint) * networkFee,
```

Comment:
Can't this overflow as well?

Reply:
No. Each of `currPoint` and `prevPoint` is a square root of some value, hence at most 128 bits.
Hence their difference is at most 128 bits.
And since `networkFee` is at most 32 bits, the product above is at most 160 bits.